stores/wallet.go

```diff
@@ -88,11 +89,22 @@ func (s *SQLStore) UnspentSiacoinElements() ([]wallet.SiacoinElement, error) {
 }
 
 // Transactions implements wallet.SingleAddressStore.
-func (s *SQLStore) Transactions(since time.Time, max int) ([]wallet.Transaction, error) {
+func (s *SQLStore) Transactions(before, since time.Time, offset, limit int) ([]wallet.Transaction, error) {
+	beforeX := int64(math.MaxInt64)
+	sinceX := int64(0)
+	if !before.IsZero() {
+		beforeX = before.Unix()
+	}
+	if !since.IsZero() {
+		sinceX = since.Unix()
+	}
+	if limit == 0 || limit == -1 {
+		limit = math.MaxInt64
+	}
+
 	var dbTxns []dbTransaction
-	err := s.db.Find(&dbTxns).
-		Where("timestamp > ?", since.UnixNano()).
-		Limit(max).
+	err := s.db.Raw("SELECT * FROM transactions WHERE timestamp > ? AND timestamp < ? ORDER BY timestamp DESC LIMIT ? OFFSET ?",
```

Comment:
Do we need to `>=` on one side to page properly? e.g. if you range over a date range with
`since x && before y` you need `timestamp >= && timestamp <` I think.

Reply:
Pagination is done through `offset` and `limit`. Because you usually want to fetch equal size batches to fill your UI or whatever. The timestamps can't give you that so they merely act as filters.

Reply:
What if I want to show transaction per day? I see your point but I also don't see why we wouldn't just change it to `>=` so you can properly fetch transaction for consecutive date ranges.